target-gen/src/commands/test.rs

```diff
@@ -0,0 +1,129 @@
+use std::path::Path;
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+
+use anyhow::Result;
+use colored::Colorize;
+use indicatif::{MultiProgress, ProgressBar, ProgressStyle};
+use probe_rs::{
+    flashing::{DownloadOptions, FlashLoader, FlashProgress},
+    Permissions, Session,
+};
+
+use super::export::{cmd_export, DEFINITION_EXPORT_PATH};
+
+pub fn cmd_test() -> Result<()> {
+    // Generate the binary
+    cmd_export()?;
+
+    probe_rs::config::add_target_from_yaml(Path::new(DEFINITION_EXPORT_PATH))?;
+    let mut session =
+        probe_rs::Session::auto_attach("algorithm-test", Permissions::new().allow_erase_all())?;
+
+    let mut loader = session.target().flash_loader();
+    let data = (0..0x401)
+        .into_iter()
+        .map(|n| (n % 256) as u8)
+        .collect::<Vec<_>>();
+    loader.add_data(0x0, &data)?;
+
+    run_flash_download(&mut session, loader, false)?;
+
+    Ok(())
+}
+
+/// Performs the flash download with the given loader. Ensure that the loader has the data to load already stored.
+/// This function also manages the update and display of progress bars.
+pub fn run_flash_download(
+    session: &mut Session,
+    loader: FlashLoader,
+    do_chip_erase: bool,
+) -> Result<()> {
+    // Start timer.
+    let instant = Instant::now();
+
+    let mut download_option = DownloadOptions::default();
+    download_option.keep_unwritten_bytes = false;
+    download_option.do_chip_erase = do_chip_erase;
+    download_option.disable_double_buffering = true;
```

Comment:
This should be configurable, otherwise it's hard to test.

Reply:
I did not feed out this option but implemented different tests in the command directly :)